spython/main/parse/parsers/docker.py

```diff
@@ -9,25 +9,57 @@
 import os
 import re
 
-from .environment import parse_env
-from .recipe import Recipe
 from spython.logger import bot
+from .base import ParserBase
 
-class DockerRecipe(Recipe):
 
-    def __init__(self, recipe=None):
-        '''a Docker recipe parses a Docker Recipe into the expected fields of
-           labels, environment, and install/runtime commands. We save working
-           directory as we parse, and the last one can be added to the runscript
-           of a Singularity recipe.
+class DockerParser(ParserBase):
+
+    name = 'docker'
+
+    def __init__(self, recipe='Dockerfile', load=True):
+        '''a docker parser will read in a Dockerfile and put it into a Recipe
+           object.
 
            Parameters
            ==========
-           recipe: the recipe file (Dockerfile) to parse
+           recipe: the Dockerfile to parse. If not defined, deafults to 
+                   Dockerfile assumed to be in the $PWD.
 
         '''
-        self.name = "docker"
-        super(DockerRecipe, self).__init__(recipe)
+        super(DockerParser, self).__init__(recipe, load)
```

Comment:
Ouch. Base class does nothing on default and subclass uses a default file? Just require a file in all cases

Reply:
The "Ouch" is not necessary.

Reply:
And the base class does quite a bit, actually:

```python
    def __init__(self, filename=None, load=True):
        '''a generic recipe parser holds the original file, and provides 
           shared functions for interacting with files. If the subclass has
           a parse function defined, we parse the filename

           Parameters
           ==========
           recipe: the recipe file to parse.

        '''
        self.filename = filename
        self._run_checks()
        self.recipe = Recipe(self.filename)

        if self.filename:

            # Read in the raw lines of the file
            self.lines = read_file(self.filename)

            # If parsing function defined, parse the recipe
            if load is True and hasattr(self, 'parse'):
                self.parse()

```

Reply:
I wrote "on default". So `parser = ParserBase()` does nothing but `parser = DockerParser()` loads a file. The linter warning is actually a good one to show you that something "strange" is going on. In this case: Different default arguments